src/Client/MultiplexedConnections.cpp

```diff
@@ -210,6 +217,7 @@ Packet MultiplexedConnections::drain()
 
         switch (packet.type)
         {
+            case Protocol::Server::ReadTaskRequest:
```

Comment:
Now I try to understand what happen if reader sent `ReadTaskRequest`, but connection has already sent cancel.
Should we also process cancel packet while waiting for `ReadTaskResponse`?

Reply:
I though about it, I will try to fix.

Reply:
Now we check if `Cancel` packet was received instead of `ReadTaskResponse`